### ISSUE TITLE
customization of the dashboard

### ISSUE THREAD

**Issue by oholimoli** [performancecopilot/grafana-pcp#167]
Hi, I don't know where I can ask this question and hope somebody can give me a hint into the right direction.

The "host overview" dashboard is working and it displays all kind of different metrics very nicely.

Now I would like to customize this dashboard.

I would like to add the free disk space of logical volumes from an LVM partition.
The default dashboard shows the free disk space for sda and sdb: (`disk.dev.avactive{hostname == "$host"}`) <= what is happening here?

Currently this is a secret language to me. Where I can't find a documentation that helps somebody who is new to pcp/pmseries/redis. The `pmseries` manual is not helping much.

**Comment by natoscott** [comment]
| The default dashboard shows the free disk space for sda and sdb: (disk.dev.avactive[...]

This metric (disk.dev.avactive) is not related to free disk space.  You want one/more of these metrics:

```
$ pminfo -t filesys.full filesys.free filesys.used filesys.capacity
filesys.full [Percentage of filesystem in use]
filesys.free [Total space free on mounted filesystem (Kbytes)]
filesys.used [Total space used on mounted filesystem (Kbytes)]
filesys.capacity [Total capacity of mounted filesystem (Kbytes)]
```

Anyway, the query:  disk.dev.avactive{hostname == "$host"}

Is saying "find time series for the PCP metric named disk.dev.active having a "hostname" label, where the value matches $host (the value of $host is filled via Grafana templating, but defaults to the host running pmproxy IIRC).

| Currently this is a secret language to me

It's combining three separate things, which may make it somewhat confusing I guess.

It requires an understanding of the PCP metric naming (hierarchical namespace, explore with pminfo as above), then the pmseries(1) language itself - https://man7.org/linux/man-pages/man1/pmseries.1.html - and finally an understanding of Grafana templating as described here - https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html

**Comment by oholimoli** [comment]
thank you for your answer. I will try to add `filesys.capacity/used/free` since this metric is not logged by default :)

**Comment by natoscott** [comment]
@oholimoli no problem - you might have some joy trying the pcp-zeroconf package (deb/rpm), this enables logging of more metrics 'by default', including these filesystem metrics.

**Comment by oholimoli** [comment]
@natoscott thank you :), I installed pcp-zeroconf but there was also no `filesys.capacity.*` available. But what I'm using now is `filesys.full` this metric is also logged in normal pcp installation.

I added this metric to the overview section. I think it would make sense to add it to the dashboard template.

**Comment by oholimoli** [comment]
What I still find confusing why this data source is called PCP Redis? Is it necessary to install Redis?

**Comment by andreasgerstmayr** [comment]
Hi @oholimoli, grafana-pcp comes with three datasources: https://grafana-pcp.readthedocs.io/en/latest/datasources/index.html

If you want to use live metrics (i.e. when you close and open your browser, the metrics history is gone), you can use PCP Vector. Otherwise, the metrics need to be stored somewhere for fast retrieval, and in that case, you need to install Redis and use the PCP Redis datasource.

From your initial question it looks like you're already using PCP Redis, so I'm a bit confused now, do you have Redis installed?

**Comment by oholimoli** [comment]
yes, I installed Redis but I did not understand where the configuration is done to work together with PCP

**Comment by andreasgerstmayr** [comment]
Per default Redis listens on localhost:6379, and pmproxy is configured to automatically use this Redis instance (if available). Additional Redis-related settings are in the pmproxy configuration file at `/etc/pcp/pmproxy/pmproxy.conf`

**Comment by natoscott** [comment]
You can see which metrics are indexed in Redis using:

```
pmseries -m | grep filesys
```

This doc may also help re Redis + PCP deployment options:
https://pcp.readthedocs.io/en/latest/HowTos/scaling/index.html
